inc/classes/preload/class-abstract-process.php

```diff
@@ -0,0 +1,213 @@
+<?php
+namespace WP_Rocket\Preload;
+
+/**
+ * Abstract class to be extended by preload process classes.
+ * Extends the background process class for the preload background process.
+ * The class extending this one must have a $action property and a task() method at least.
+ *
+ * @since  3.6
+ * @author Grégory Viguier
+ *
+ * @see WP_Background_Process
+ */
+abstract class Process extends \WP_Background_Process {
+	/**
+	 * Prefix
+	 *
+	 * @since  3.2
+	 * @var    string
+	 * @author Remy Perona
+	 */
+	protected $prefix = 'rocket';
+
+	/**
+	 * Format the item to an array.
+	 *
+	 * @since  3.6
+	 * @access public
+	 * @author Grégory Viguier
+	 *
+	 * @param  array|string $item {
+	 *     The item to preload: an array containing the following values.
+	 *     A string is allowed for backward compatibility (for the URL).
+	 *
+	 *     @type string $url    The URL to preload.
+	 *     @type bool   $mobile True when we want to send a "mobile" user agent with the request. Optional.
+	 * }
+	 * @return array|bool The formatted item. False for invalid items.
+	 */
+	public function format_item( $item ) {
+		if ( is_string( $item ) ) {
+			$item = [
+				'url' => $item,
+			];
+		} elseif ( ! is_array( $item ) ) {
+			return false;
+		}
+
+		if ( empty( $item['url'] ) ) {
+			return false;
+		}
+
+		$item['mobile'] = ! empty( $item['mobile'] );
+
+		return $item;
+	}
+
+	/**
+	 * Tell if mobile preload is enabled.
+	 *
+	 * @since  3.6
+	 * @access public
+	 * @author Grégory Viguier
+	 *
+	 * @return bool
+	 */
+	public function is_mobile_preload_enabled() {
+		$enabled = get_rocket_option( 'manual_preload' ) && get_rocket_option( 'cache_mobile' ) && get_rocket_option( 'do_caching_mobile_files' );
+
+		/**
+		 * Tell if mobile preload is enabled.
+		 *
+		 * @since  3.6
+		 * @author Grégory Viguier
+		 *
+		 * @param bool   $enabled True when enabled. False otherwise.
+		 * @param string $action  Specific action identifier for the current preload type.
+		 */
+		return (bool) apply_filters( 'rocket_mobile_preload_enabled', $enabled, $this->action );
+	}
+
+	/**
+	 * Get the user agent to use for the item.
+	 *
+	 * @since  3.6
+	 * @access public
+	 * @author Grégory Viguier
+	 *
+	 * @param  array $item {
+	 *     The item to preload: an array containing the following values.
+	 *
+	 *     @type string $url    The URL to preload.
+	 *     @type bool   $mobile True when we want to send a "mobile" user agent with the request. Optional.
+	 * }
+	 * @return string
+	 */
+	public function get_item_user_agent( $item = [] ) {
+		if ( $item['mobile'] ) {
+			return 'WP Rocket/Preload iPhone';
```

Comment:
Why iPhone?

Reply:
Because it’s the [first value in the mobile devices](https://github.com/wp-media/wp-rocket/blob/a37e46f58259e38dbf895061536551763f8aace7/inc/vendors/classes/class-rocket-mobile-detect.php#L172).

Reply:
To clarify (sorry I wasn't clear enough), this is the library WPR uses to detect mobile devices. It is used to separate mobile and non mobile caches, so I use the same source for my tests.